django_rest_generator/parser/__init__.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2022 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from dataclasses import dataclass, make_dataclass
+from typing import List
+from prance import ResolvingParser
+from prance.util.resolver import RESOLVE_HTTP, RESOLVE_FILES
+from collections import defaultdict
+from django_rest_generator.parser.models import (
+    Schema,
+    CommonDataclass,
+    EndpointOperation,
+    Endpoint,
+    Resource,
+)
+from django_rest_generator.utils import find_nested_keys
+
+
+@dataclass
+class OpenAPISpec:
+    schemas: List[Schema]
+    resources: List[Resource]
+
+    @staticmethod
+    def _parse_resources_from_openapi(specification, server_base):
+        resources = defaultdict(lambda: defaultdict(list))
```

Comment:
this method is a bit long

is it possible to break it down a bit

Reply:
its really doing 2 distinct things though

its finding the resources available in the specification
and then for those resources, it goes through and generates the actual resources and endpoints.
they feel very distinct in this case "_parse_specification_resources" and "_generate_resources" or something

Reply:
i would expect it would be easier and more clear to test as well and i think it would help a future user to know what to expect at each point, in the case of debugging issues